Bang/Game/Gameplay.cs

```diff
@@ -50,8 +50,11 @@ public void DealFirstCards()
 
         public bool Defense(Player player, BangGameCard card, BangGameCard secondCard = null)
         {
-            state = state.ApplyReplyAction(card, secondCard);
-
+            State = State switch
+            {
+                WaitingMissedCardsAfterGatlingState _ => State.ApplyReplyAction(player, card, secondCard),
+                _ => State.ApplyReplyAction(card, secondCard),
```

Comment:
I don't like this. I suggest 
to remove `ApplyReplyAction(BangGameCard, BangGameCard)` everywhere, and use `ApplyReplyAction(Player, BangGameCard, BangGameCard)` instead of

Reply:
I wanted to play with a new for me C# feature. But should agree that your suggestion is better